theseus/labs/lie/lie_tensor.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from enum import Enum
+from typing import List, Tuple, Union
+
+import torch
+
+from theseus.labs.lie.functional.lie_group import UnaryOperatorOpFnType
+from theseus.labs.lie.functional import se3 as _se3_base
+from theseus.labs.lie.functional import so3 as _so3_base
+
+
+TensorType = Union[torch.Tensor, "LieTensor"]
+
+
+class LieType(Enum):
+    SE3 = 0
+    SO3 = 1
+
+
+SE3 = LieType.SE3
+SO3 = LieType.SO3
+
+_JFnReturnType = Tuple[List[torch.Tensor], TensorType]
+
+
+class LieTensor:
+    def __init__(self, data_tensor: torch.Tensor, ltype: LieType):
```

Comment:
In terms of usage what is `ltype` passed as and printed as?
Can I do `x = LieTensor(data, 'SE3')` or is the correct way to use `x = LieTensor(data, lie_tensor.SE3)`
And what prints when I do `print(x.ltype)`, is it 0 from the Enum?

Maybe we add some convenience stuff if needed in `lie/__init__.py`.

Reply:
It should be printing the string, but I forgot to add `self.ltype.name` to the format. 

To be consistent with torch, I'd prefer if people do `x = LieTensor(data, ltype=lie.SE3)` and disallow strings.

Reply:
Could `lie.SE3` for type use cause confusion with any functional use via `lie.se3.` or `lie.SE3.` I don't think we have any functions in the latter namespace but do we expect any later?

Reply:
See [this](https://github.com/facebookresearch/theseus/pull/450#discussion_r1088341427) comment.